system_scripts/subiquity-legacy-cloud-init-validate

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Legacy script for compatibility on systems where the 'cloud-init schema'
+is unsupported.
+"""
+
+import argparse
+import sys
+from typing import Any, Dict
+
+import yaml
+from cloudinit import safeyaml
+from cloudinit.config.schema import (
+    SchemaValidationError,
+    get_schema,
+    validate_cloudconfig_schema,
+)
+
+try:
+    from cloudinit.config.schema import SchemaProblem
+except ImportError:
+
+    def SchemaProblem(x, y):
+        return (x, y)  # (not available before cloud-init 22.3)
+
+
+def validate(data: Dict[str, Any], data_source: str) -> Dict[str, str]:
+    """Validate data config adheres to strict cloud-config schema
```

Comment:
"validate that `data` adheres to strict cloud-config schema" maybe?

Reply:
Yeah that sounds better to me. Updated.